support/cas-server-support-ldap/src/main/java/org/apereo/cas/config/LdapPasswordSynchronizationConfiguration.java

```diff
@@ -50,21 +51,24 @@ protected void destroyInstance(final List list) {
 
     @ConditionalOnMissingBean(name = "ldapPasswordSynchronizationAuthenticationEventExecutionPlanConfigurer")
     @Bean
-    @SneakyThrows
     @Autowired
     public AuthenticationEventExecutionPlanConfigurer ldapPasswordSynchronizationAuthenticationEventExecutionPlanConfigurer(
         @Qualifier("ldapPasswordSynchronizationAuthenticationPostProcessorListFactoryBean")
         final ListFactoryBean ldapPasswordSynchronizationAuthenticationPostProcessorListFactoryBean) {
-        val postProcessorList = ldapPasswordSynchronizationAuthenticationPostProcessorListFactoryBean.getObject();
         return plan -> {
-            val ldap = casProperties.getAuthn().getPasswordSync().getLdap();
-            ldap.stream()
-                .filter(LdapPasswordSynchronizationProperties::isEnabled)
-                .forEach(instance -> {
-                    val postProcessor = new LdapPasswordSynchronizationAuthenticationPostProcessor(instance);
-                    postProcessorList.add(postProcessor);
-                    plan.registerAuthenticationPostProcessor(postProcessor);
-                });
+            try {
+                val postProcessorList = Objects.requireNonNull(ldapPasswordSynchronizationAuthenticationPostProcessorListFactoryBean.getObject());
+                val ldap = casProperties.getAuthn().getPasswordSync().getLdap();
+                ldap.stream()
+                        .filter(LdapPasswordSynchronizationProperties::isEnabled)
+                        .forEach(instance -> {
+                            val postProcessor = new LdapPasswordSynchronizationAuthenticationPostProcessor(instance);
+                            postProcessorList.add(postProcessor);
+                            plan.registerAuthenticationPostProcessor(postProcessor);
+                        });
+            } catch (final Exception e) {
+                throw new BeanCreationException("Error creating ldapPasswordSynchronizationAuthenticationPostProcessorListFactoryBean: {}", e.getMessage(), e);
```

Comment:
Also the exception message is misleading; there is nothing that would indicate the problem might have to do with `ldapPasswordSynchronizationAuthenticationPostProcessorListFactoryBean`, if for example, something in the constructor of `LdapPasswordSynchronizationAuthenticationPostProcessor` goes wrong.

Reply:
You might want to reference `ldapPasswordSynchronizationAuthenticationEventExecutionPlanConfigurer` for the time being as the bean whose creation attempt has failed.